spec/latest/json-ld/index.html

```diff
@@ -972,7 +972,7 @@ <h2>JSON-LD 1.1 Processing Mode</h2>
   </pre>
 
   <p>The first <code>context</code> encountered when processing a
-    document determines the <code>processing mode</code>,
+    document which contains <code>@version</code> determines the <code>processing mode</code>,
```

Comment:
This works for the current case of 1.0 being undeclared explicitly, AFAICT, as only 1.1 will be encountered, and thus it will necessarily be encountered first. If 1.0 was declared explicitly, then the 1.1 features would not be available for the scenario described in the issue.

Would it be possible to instead set it to the highest version encountered as contexts are processed, so long as the major version is the same?  Then a set of 1.1, 1.0, 1.2 would resolve until 1.2 processing, not 1.1. If a new major version is encountered, then it should probably be an error state to avoid the backwards incompatible changes.

Reply:
The only way to set version to 1.1 is through the API; the idea was to use this as a way detect the situation of using 1.1 inadvertently. By not setting it, it is determined by the version `@version` seen, and presumed to be 1.0, if nothing is ever seen.

You raise a good point about upward compatibility, and it's not unreasonable to do that, if we don't care about detecting the use of a higher version.

In Ruby, I can declare dependencies on version numbers such as `"=1.0"`, `"~>1.0"` or `">= 1.0"`, which allow exactly 1.0, anything in 1.x which is at least 1.0 or anything greater than 1.0 respectively.

The question is, it useful to set the processor up to be restricted to 1.0 only, treat everything as 1.1, or resolve on the first/greatest value of `@version`?

Reply:
When 1.2 comes about (if ever), it will need to determine how to handle encountering `@version: 1.1`. Right now, the only valid value of `@version` is 1.1, so for this spec, it can't happen any other way, other than through the API `processingMode` option.

Reply:
Agreed, and we might need to change it again anyway in that hypothetical future version.